src/pages/VestingClaim.tsx

```diff
@@ -0,0 +1,191 @@
+        SDL_TOKEN_ABI,
+        provider.getSigner(),
+      ) as Sdl
+      const vestingContractDeployedFilter = SDLContract.filters.VestingContractDeployed(
+        null,
+        null,
+      )
+      try {
+        const events = await SDLContract.queryFilter(
+          vestingContractDeployedFilter,
+        )
+        const currentBeneficiary = events.filter(
+          (event) => event.args?.beneficiary === account,
+        )
+        setIsValidBene(currentBeneficiary.length > 0)
+        if (isValidBene) {
+          const vestingContract = new EthersContract(
+            currentBeneficiary[0].args?.vestingContract,
+            INVESTOR_EMPLOYEE_VESTING_CONTRACT_ABI,
+            provider.getSigner(),
+          ) as Vesting
+          setVestingContract(vestingContract)
+          try {
+            const remainingAmount = commify(
+              formatBNToString(
+                await SDLContract.balanceOf(vestingContract.address),
+                18,
+                2,
+              ),
+            )
+            setRemainingAmount(remainingAmount)
+          } catch (err) {
+            console.error(err)
+            notifyCustomError({
+              ...(err as Error),
+              message: "Unable to get total pending amount",
+            })
+          }
+          try {
+            const claimableVestedAmount = commify(
+              formatBNToString(await vestingContract.vestedAmount(), 18, 2),
+            )
+            setClaimableVestedAmount(claimableVestedAmount)
+          } catch (err) {
+            console.error(err)
+            notifyCustomError({
+              ...(err as Error),
+              message: "Unable to get vested amount",
+            })
+          }
+        }
+      } catch (err) {
+        console.error(err)
+        notifyCustomError({
+          ...(err as Error),
+          message: "Unable to query for vesting contracts",
+        })
+      }
+    }
+    void fetchBeneficiaries()
+  }, [account, chainId, library, isValidBene])
+
+  const onClaimClick: () => void = async () => {
+    if (!vestingContract) return
+    try {
+      const txn = await vestingContract.release()
+      notifyHandler(txn?.hash, "claim")
+      await txn?.wait()
+      setClaimableVestedAmount("0.0")
+    } catch (err) {
+      console.error(err)
+      notifyCustomError({
+        ...(err as Error),
+        message: "Unable to claim vested tokens",
+      })
+    }
+  }
+
+  return (
+    <>
+      <TopMenu activeTab={t("n/a")} />
+      <div className={styles.container}>
+        <div className={styles.gradient}></div>
+        <div className={styles.logoWrapper}>
+          <div className={styles.logo}>
+            <img src={logo} />
+          </div>
+        </div>
+        <div className={styles.mainContent}>
+          {isValidBene ? (
+            <>
+              <div className={styles.tokenBalance}>
+                {claimableVestedAmount} of {remainingAmount} remaining tokens
```

Comment:
TODO: translate. grep for `stepAOfB` for how to insert variables into translations

Reply:
Should all the displayed text be translated?

Reply:
I'm not actually sure how many of our investors are ZH-first/only, translating is probably not a high-pri. It is necessary for public-facing features though.